apps/omg_child_chain/test/omg_child_chain/event_fetcher_test.exs

```diff
@@ -0,0 +1,497 @@
+      # this deposit will get called out below
+      {from_block + 1, get_signature_from_event(events, :deposit_created), [deposit_created_2]},
+      {from_block + 1, get_signature_from_event(events, :in_flight_exit_output_piggybacked), []},
+      {from_block + 1, get_signature_from_event(events, :in_flight_exit_started), []},
+      {from_block + 1, get_signature_from_event(events, :in_flight_exit_input_piggybacked), []},
+      {from_block + 1, get_signature_from_event(events, :exit_started), []},
+      {to_block, get_signature_from_event(events, :deposit_created), []},
+      {to_block, get_signature_from_event(events, :in_flight_exit_output_piggybacked), []},
+      {to_block, get_signature_from_event(events, :exit_started), [exit_started_log]},
+      {to_block, get_signature_from_event(events, :in_flight_exit_input_piggybacked),
+       [in_flight_exit_input_piggybacked_log]},
+      {to_block, get_signature_from_event(events, :in_flight_exit_started), []}
+    ]
+
+    _ = :ets.insert(table, data)
+
+    defmodule test_name do
+      alias OMG.Watcher.EventFetcherTest
+
+      def get_ethereum_events(_from_block, _to_block, _signatures, _contracts) do
+        {:ok,
+         [
+           %{
+             amount: 10,
+             blknum: 1,
+             currency: <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>,
+             eth_height: 1,
+             event_signature: "DepositCreated(address,uint256,address,uint256)",
+             log_index: 0,
+             owner: <<59, 159, 76, 29, 210, 110, 11, 229, 147, 55, 59, 29, 54, 206, 226, 0, 140, 190, 184, 55>>,
+             root_chain_txhash:
+               <<77, 114, 166, 63, 244, 47, 29, 181, 10, 242, 195, 110, 139, 49, 65, 1, 210, 254, 163, 224, 0, 53, 117,
+                 243, 2, 152, 233, 21, 63, 227, 216, 238>>
+           },
+           %{
+             amount: 10,
+             blknum: 1,
+             currency: <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>,
+             eth_height: 2,
+             event_signature: "DepositCreated(address,uint256,address,uint256)",
+             log_index: 0,
+             owner: <<59, 159, 76, 29, 210, 110, 11, 229, 147, 55, 59, 29, 54, 206, 226, 0, 140, 190, 184, 55>>,
+             root_chain_txhash:
+               <<77, 114, 166, 63, 244, 47, 29, 181, 10, 242, 195, 110, 139, 49, 65, 1, 210, 254, 163, 224, 0, 53, 117,
+                 243, 2, 152, 233, 21, 63, 227, 216, 238>>
+           }
+         ]}
+      end
+
+      def get_call_data(_tx_hash) do
+        {:ok, EventFetcherTest.start_standard_exit_log()}
+      end
+    end
+
+    :sys.replace_state(event_fetcher_name, fn state -> Map.put(state, :rpc, test_name) end)
+
+    {:ok, data} = EventFetcher.deposit_created(event_fetcher_name, from_block, to_block)
+    assert Enum.at(data, 0) == deposit_created
+    assert Enum.at(data, 1) == deposit_created_2
+  end
+
```

Comment:
pls add a test for the `missing_blocks`  capabilities

Reply:
pls take a look at the comment above https://github.com/omisego/elixir-omg/pull/1376#discussion_r389799976

every time you "make a call" into the future, as in, ets is empty for particular blocks, the _missing blocks_ capabilities are invoked

Reply:
ah, gotcha, I don't know where I got the presumption that the `EventFetcher` fetches pro-actively from. Maybe from the `Fetcher` in the name? What it does is it just opportunistically pools together the event requests across all topics, right. Maybe `EthereumEventAggregator`? but no strong opinion here.

Reply:
I did try the pro-active implementation but there was no benefit (the downside was the increased complexity really)